### PR TITLE
Add remaining Lavaland loot, kinetic accelerator refactor

### PR DESCRIPTION
**Pull request by Markolie**
Pretty much all these changes come from /tg/. 

Adds the remaining Lavaland loot from Tendrils. Most of these were developed by @KorPhaeron I believe. Let me know if I need to credit anyone else.
* Eye of God: Gives you xray vision, lets you see in the dark and lets you scan reagents. An all-in-one eyepiece.
* Inquisitor/Berserker hardsuits: Basically intended as a well-armored hardsuit you can find on Lavaland. They share stats with ERT hardsuits.
* AoE kinetic accelerator modkit: Causes the kinetic accelerator to become AoE.
* Paradox Bag: A set of bags that share an inventory.
* Lava Boat: A boat you can use to sail over lava. Comes from the "ship in a bottle" item. Does not properly protect you against our current (unsimulated) lava, but it will once we implement the rest of Lavaland.
* Wisp Lantern: A lantern with a wisp in it that illuminates the area around you and gives you thermal vision.
* Warp Cube: A set of cubes that lets the user teleport from one cube to the other.
* Meat Hook: A hook "gun" you can use to pull a mob towards you.
* Immortality Talisman: When activated, this talisman makes you immune to all damage for a short while.

This PR also refactors the kinetic accelerator to use modkits instead of subtypes (by @KorPhaeron): https://github.com/tgstation/tgstation/pull/19870 and https://github.com/tgstation/tgstation/pull/19942.  In short, this adds 6 mod slots to the KA that modifies its effects (such as more damage indoor, more range, shorter cooldown).

I've also added a number of helpful defines and a dir helper proc that we should've ported from /tg/ ages ago. I've only added them: the code still uses the magic numbers/direct dir setting. At some point we should run a find/replace in the code to update these. These make porting things from /tg/ much easier though.
* Added inventory weight defines. For instance, instead of using 1 for tiny items, you can (and should) now use WEIGHT_CLASS_TINY.
* Added defines for every layer.
* Added a setDir() proc which should be used instead of setting a dir directly.

Additional changes:
* Renamed the "F" variable on guns to gun_light.
* Removed the sharp/edge variables from the SORD so it's actually useless.
* Added a new damage type for suicides that don't actually work: "SHAME". Currently used by the SORD. 
* I've split up the megafauna loot into separate files.
* Megafauna no longer regen health while on station so they aren't as much as an issue if they are ever transported there.
* Moved the reagent liquid/gas/solid defines to its own file so it can be accessed anywhere in the code.
* Hierophant runes can not be brought to Central Command.
* You can't put your own name on a blood contract.
* Adds three new variable to soul stones: "usability" which prevents the stone from being used by anyone but cultists/wizards, "reusable" which means the soul stone can only be used once and "spent" which means the soul stone has been used. Also adds some helpful text for cultists/observers/wizards when examining constructs, and some updated text the shade receives depending on the summoner being a cultist or wizard.

I've also added the potion of flight, but as we don't have the angel race or support for wings so the potion doesn't do anything. If someone more familiar with species/accessory code feels like porting this, please do!

🆑 Markolie
rscadd: Ports over the remaining Lavaland loot items from /tg/ (most developed by KorPhaeron). 
rscadd: Refactors the kinetic accelerator to use modkits instead of subtypes: these unique mods (such as more range or shorter cooldown) can be purchased from the mining vendor or built by R&D or robotics (developed by KorPhaeron). 
/🆑

### CODE REVIEW

File: code/game/atoms.dm [hunk 1]
```diff
@@ -81,6 +81,10 @@
 		reagents = null
 	invisibility = 101
 	return ..()
+	
+//Hook for running code when a dir change occurs
+/atom/proc/setDir(newdir)
```
Comment (Fox-McCloud) [inline code/game/atoms.dm]: You'll want to get rid of the old version of this, which is `set_dir`

File: code/game/objects/items/weapons/weaponry.dm [hunk 2]
```diff
@@ -23,18 +23,14 @@
...
-		to_chat(viewers(user), "<span class='suicide'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
-		return(BRUTELOSS)
-
-/obj/item/weapon/sord/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
-	playsound(loc, 'sound/weapons/bladeslice.ogg', 50, 1, -1)
-	return ..()
+/obj/item/weapon/sord/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] is trying to impale themself with [src]! It might be a suicide attempt if it weren't so shitty.</span>", \
+	"<span class='suicide'>You try to impale yourself with [src], but it's USELESS...</span>")
+	return SHAME
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Oh boy.

File: code/modules/mining/lavaland/loot/tendril_loot.dm [hunk 3]
```diff
@@ -0,0 +1,379 @@
...
+
+/datum/reagent/flightpotion
+	name = "Flight Potion"
+	id = "flightpotion"
+	description = "Strange mutagenic compound of unknown origins."
+	reagent_state = LIQUID
+	color = "#FFEBEB"
+
+/datum/reagent/flightpotion/reaction_mob(mob/living/M, method = TOUCH, reac_volume, show_message = 1)
+	to_chat(M, "<span class='warning'>This item is currently non-functional.</span>")
```
Comment (Fox-McCloud) [inline code/modules/mining/lavaland/loot/tendril_loot.dm]: ahahah.

File: code/modules/mining/lavaland/loot/tendril_loot.dm [hunk 4]
```diff
@@ -0,0 +1,379 @@
...
+	..()
+	wisp = new(src)
+
+/obj/item/device/wisp_lantern/Destroy()
+	if(wisp)
+		if(wisp.loc == src)
+			qdel(wisp)
+		else
+			wisp.visible_message("<span class='notice'>[wisp] has a sad feeling for a moment, then it passes.</span>")
+	..()
```
Comment (Fox-McCloud) [inline code/modules/mining/lavaland/loot/tendril_loot.dm]: `return ..()`

File: code/game/atoms.dm (hunk 1 above)
Comment (Markolie) [inline code/game/atoms.dm]: Wait, we already have a proc for this? I'll take care of that.

File: code/modules/projectiles/guns/energy/special.dm [hunk 5]
```diff
@@ -84,141 +84,6 @@
 	ammo_type = list(/obj/item/ammo_casing/energy/mindflayer)
 	ammo_x_offset = 2
 
-/obj/item/weapon/gun/energy/kinetic_accelerator
-	name = "proto-kinetic accelerator"
-	desc = "In the year 2544, only a year after the discovery of a potentially \
```
Comment (Markolie) [inline code/modules/projectiles/guns/energy/special.dm]: RIP super long description

File: code/modules/projectiles/projectile.dm [hunk 6]
```diff
@@ -52,7 +52,7 @@
 	var/forcedodge = 0 //to pass through everything
 	var/dismemberment = 0 //The higher the number, the greater the bonus to dismembering. 0 will not dismember at all.
 
-	var/log = 1 //whether print to admin attack logs or just keep it in the diary
+	var/log_override = FALSE //whether print to admin attack logs or just keep it in the diary
```
Comment (Markolie) [inline code/modules/projectiles/projectile.dm]: I renamed this variable so it matches /tg/'s version.

File: code/game/atoms.dm (hunk 1 above)
Comment (tigercat2000) [inline code/game/atoms.dm]: Or, we can _not_ rename a proc, and just mass-replace everything contained in here with `set_dir` instead of `setDir` 😛

Comment (Markolie) [inline code/game/atoms.dm]: Ironically only about four files use set_dir (aka almost none). Considering the amount of stuff we port from /tg/, I think for the easy of -stealing- copying code we should use their proc/variable names.

File: code/__DEFINES/layers.dm [hunk 7]
```diff
@@ -0,0 +1,73 @@
+//Defines for atom layers and planes
+//KEEP THESE IN A NICE ACSCENDING ORDER, PLEASE
+
+#define CLICKCATCHER_PLANE -99
```
Comment (tigercat2000) [inline code/__DEFINES/layers.dm]: This is already defined in `code/_onclick/_defines.dm`

File: code/__DEFINES/layers.dm [hunk 8]
```diff
@@ -0,0 +1,73 @@
...
+//HUD layer defines
+
+#define FULLSCREEN_PLANE 18
+#define FLASH_LAYER 18
+#define FULLSCREEN_LAYER 18.1
+#define UI_DAMAGE_LAYER 18.2
+#define BLIND_LAYER 18.3
+#define CRIT_LAYER 18.4
+
+#define HUD_PLANE 19
```
Comment (tigercat2000) [inline code/__DEFINES/layers.dm]: This is also defined in `code/_onclick/_defines.dm`

File: code/game/atoms.dm (hunk 1 above)
Comment (Fox-McCloud) [inline code/game/atoms.dm]: @tigercat2000 he already replaced all the instances of `set_dir` with `setDir`...andd, well, given what @Markolie stated, I'm inclined that keeping our proc names the same as TG's is more beneficial to us and future coders.

File: code/__DEFINES/layers.dm [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+#define AREA_LAYER 10
+#define MASSIVE_OBJ_LAYER 11
+#define POINT_LAYER 12
+
+#define LIGHTING_PLANE 15
+#define LIGHTING_LAYER 15
+
+//HUD layer defines
+
+#define FULLSCREEN_PLANE 18
```
Comment (tigercat2000) [inline code/__DEFINES/layers.dm]: You should probably actually use all of these hud ones at least if you are going to include them...

Comment (Markolie) [inline code/__DEFINES/layers.dm]: Ideally we should run a find-and-replace on the entire codebase to replace the inventory, layer and hud magic number in all files with these defines. However, that'd bloat the size of this PR to hundreds of files which I'd rather avoid. Not to mention that I don't trust my regex skills sufficiently to not mess everything up.

Comment (tigercat2000) [inline code/__DEFINES/layers.dm]: Well, yeah, the giant regex should be done sometime later- But you could just apply some of these, the HUD ones especially are really only defined in like 5 places 😛

Comment (Markolie) [inline code/__DEFINES/layers.dm]: Ah, I wasn't aware of that. I'll look into doing that tomorrow when I test this!

File: code/game/atoms.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/atoms.dm]: you'll use *MY* spelling damnit! :3

File: code/__DEFINES/layers.dm (hunk 9 above)
Comment (Markolie) [inline code/__DEFINES/layers.dm]: I've removed the plane defines for now until (if) we ever port https://github.com/tgstation/tgstation/pull/20707. Until then these aren't really worth defining like the layer ones are.

**Comment by Markolie** [comment]
I've also pushed some soul stone updates (in order to add the soulstone/anybody) type: 

Adds three new variable to soul stones: "usability" which prevents the stone from being used by anyone but cultists/wizards, "reusable" which means the soul stone can only be used once and "spent" which means the soul stone has been used. Also adds some helpful text for cultists/observers/wizards when examining constructs.

These variables are set to not change anything to existing soul stones though - for now this is only for adminbus shenanigans. If we ever want to nerf cult constructs this'd be the way to go.

**Comment by Markolie** [comment]
I'm done testing! We should be all good to go on this one.

File: code/modules/projectiles/gun.dm [hunk 10]
```diff
@@ -284,7 +284,7 @@ obj/item/weapon/gun/proc/newshot()
 	if(!isturf(user.loc))
 		to_chat(user, "<span class='warning'>You cannot turn the light on while in this [user.loc]!</span>")
 	gun_light.on = !gun_light.on
-	to_chat(user, "<span class='notice'>You toggle the gun_light [gun_light.on ? "on":"off"].</span>")
+	to_chat(user, "<span class='notice'>You toggle the gun light [gun_light.on ? "on":"off"].</span>")
```
Comment (tigercat2000) [inline code/modules/projectiles/gun.dm]: i think this might supposed to be `[gun_light]`

Comment (Markolie) [inline code/modules/projectiles/gun.dm]: Ah no - that was an error on my end. I did a mass replace in the file and accidentally replaced this string (and the action button name) as well. This fixes that again.

**Comment by Fox-McCloud** [comment]
@Markolie 

Damn sprite conflicts.

File: code/game/objects/items/weapons/weaponry.dm [hunk 11]
```diff
@@ -23,18 +23,14 @@
...
 	w_class = 3
+	hitsound = 'sound/weapons/bladeslice.ogg'
 	attack_verb = list("attacked", "slashed", "stabbed", "sliced", "torn", "ripped", "diced", "cut")
 
-	suicide_act(mob/user)
-		to_chat(viewers(user), "<span class='suicide'>[user] is impaling \himself with the [src.name]! It looks like \he's trying to commit suicide.</span>")
-		return(BRUTELOSS)
-
-/obj/item/weapon/sord/attack(mob/living/carbon/M as mob, mob/living/carbon/user as mob)
-	playsound(loc, 'sound/weapons/bladeslice.ogg', 50, 1, -1)
```
Comment (Fox-McCloud) [inline code/game/objects/items/weapons/weaponry.dm]: Wow; these still exist...horrible.

File: code/modules/mining/equipment_locker.dm [hunk 12]
```diff
@@ -938,18 +945,18 @@
 
 //Melee
 
-/obj/item/device/mine_bot_ugprade
+/obj/item/device/mine_bot_upgrade
```
Comment (Fox-McCloud) [inline code/modules/mining/equipment_locker.dm]: `ugprade`

Welp. Good catch

**Comment by Markolie** [comment]
Fixed and added.

**Comment by LightFire53** [comment]
A meat hook gun?

Did someone say fresh meat?

**Comment by Fox-McCloud** [comment]
>Did someone say fresh meat?

Oi! I got juice on me chins.

**Comment by Spacemanspark** [comment]
CLEAAAAAAAAVE THE MEAT

**Comment by LightFire53** [comment]
Wasn't exactly the reference I was going for. Or at all,  @Spacemanspark .

Have you never played diablo?

**Comment by Spacemanspark** [comment]
Damn it.
Diablo 3, yes, but it's been a bit.

**Comment by KorPhaeron** [comment]
Closer and closer

**Comment by Crazylemon64** [comment]
aaaaaaaa

**Pull request merged**